Sources_v3/StreamChatUI/ChatChannel/ChatFileAttachmentListView+ItemView.swift

```diff
@@ -0,0 +1,179 @@
+            super.setUp()
+
+            let tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(didTapOnAttachment(_:)))
+            addGestureRecognizer(tapRecognizer)
+        }
+
+        override open func setUpLayout() {
+            spinnerAndSizeStack.addArrangedSubview(activityIndicator)
+            spinnerAndSizeStack.addArrangedSubview(fileSizeLabel)
+
+            fileNameAndSizeStack.addArrangedSubview(fileNameLabel)
+            fileNameAndSizeStack.addArrangedSubview(spinnerAndSizeStack)
+
+            addSubview(fileNameAndSizeStack)
+            addSubview(fileIconImageView)
+            addSubview(actionIconImageView)
+
+            NSLayoutConstraint.activate([
+                fileIconImageView.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+                fileIconImageView.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor),
+                fileIconImageView.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor),
+                
+                actionIconImageView.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+                actionIconImageView.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor),
+                actionIconImageView.leadingAnchor.constraint(
+                    equalToSystemSpacingAfter: fileNameAndSizeStack.trailingAnchor,
+                    multiplier: 1
+                ),
+                actionIconImageView.widthAnchor.constraint(equalToConstant: actionIconSize.width),
+                actionIconImageView.heightAnchor.constraint(equalToConstant: actionIconSize.height),
+                
+                fileNameAndSizeStack.leadingAnchor.constraint(
+                    equalToSystemSpacingAfter: fileIconImageView.trailingAnchor,
+                    multiplier: 2
+                ),
+                fileNameAndSizeStack.centerYAnchor.constraint(equalTo: centerYAnchor)
+            ])
+        }
+
+        override open func updateContent() {
+            fileNameLabel.text = attachment?.title
+            fileIconImageView.image = attachment?.fileIcon
+            fileSizeLabel.text = attachment?.fileSize
+            actionIconImageView.image = attachment?.fileAttachmentActionIcon
+
+            if case .uploading = attachment?.localState {
+                activityIndicator.startAnimating()
+            } else {
+                activityIndicator.stopAnimating()
+            }
+        }
+
+        // MARK: - Actions
+
+        @objc open func didTapOnAttachment(_ recognizer: UITapGestureRecognizer) {
+            tapHandler()
+        }
+    }
+}
+
+// MARK: - Private
+
+private extension _ChatMessageAttachment {
+    var fileIcon: UIImage? {
+        guard let file = file else { return nil }
+
+        let icon = UIImage(named: file.type.rawValue, in: .streamChatUI)
+        let fallback = UIImage(named: "generic", in: .streamChatUI)
+
+        return icon ?? fallback
```

Comment:
nit: If you do `return icon ?? UIImage(named: "generic", in: .streamChatUI)` the fallback icon won't be loaded if it's not needed :)

Reply:
not relevant any more since icons and fallback one are moved to `UIConfig` and are loaded just once